backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java

```diff
@@ -78,7 +78,33 @@ public UUID createShop(CreateShopRequest createShopRequest) {
 		}
 
 		// TODO(b/158975796): Simplify logging across all backend APIs
-		Logger.getLogger(InMemoryVirtualQueueRepository.class.getName()).log(Level.SEVERE, "Tried to fetch tokens of a shop which is not in the ACTIVE state.");
+		Logger.getLogger(InMemoryVirtualQueueRepository.class.getName()).log(Level.SEVERE,
+				"Tried to fetch tokens of a shop which is not in the ACTIVE state.");
 		return new ArrayList<>();
 	}
+
+	public Token getNewToken(UUID shopId) {
+	    if (shopMap.get(shopId).getStatus() == ShopStatus.ACTIVE) {
+              Integer newTokenNumber = shopIdToLastAllotedTokenMap.get(shopId).incrementAndGet();
+		UUID uuid = UUID.randomUUID();
+		Token newToken = new Token();
+		newToken.setTokenId(uuid);
+		newToken.setShopId(shopId);
+		newToken.setTokenNumber(newTokenNumber);
+		newToken.setStatus(Token.Status.ACTIVE);
+		tokenMap.put(uuid, newToken);
+		if (newTokenNumber != 1) {
+		    List<Token> newTokenList = shopIdToListOfTokensMap.get(shopId);
+		    newTokenList.add(newToken);
+		    shopIdToListOfTokensMap.put(shopId, newTokenList);
+		} else {
+		    List<Token> newTokenList = new ArrayList<>();
+		    newTokenList.add(newToken);
+		    shopIdToListOfTokensMap.put(shopId, newTokenList);
+		}
+		return newToken;
+	    }
+	  // TODO: Throw exception here
```

Comment:
Can you add a severe log here that the shop is not active as in getTokens method above

Reply:
Done